independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/ListExtensionsCommandHandler.java

```diff
@@ -86,29 +91,42 @@ public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws
         } catch (IOException e) {
             throw new QuarkusCommandException("Failed to determine the list of installed extensions", e);
         }
+
+        Predicate<Extension> categoryFilter;
+        if (category != null && !category.isBlank()) {
+            categoryFilter = e -> ExtensionProcessor.of(e).getCategories().contains(category);
+        } else {
+            categoryFilter = e -> true;
+        }
+
         extensions.stream()
                 .filter(e -> !ExtensionProcessor.of(e).isUnlisted())
+                .filter(categoryFilter)
+                .sorted(Comparator.comparing(e -> e.getArtifact().getArtifactId()))
                 .forEach(e -> display(log, e, installedByKey.get(toKey(e)), all, installedOnly, currentFormatter));
         final BuildTool buildTool = invocation.getQuarkusProject().getBuildTool();
         boolean isGradle = BuildTool.GRADLE.equals(buildTool) || BuildTool.GRADLE_KOTLIN_DSL.equals(buildTool);
 
-        if (!cli) {
-            if ("concise".equalsIgnoreCase(format)) {
+        if (cli && !batchMode) {
```

Comment:
Note: previously there were no hints for CLI. The hints that were deleted were in fact moved to maven and gradle plugins, and these are new ones fro CLI.

Reply:
Right. And they can go into the CLI (because the alternate ways to specify options are defined there.. ).